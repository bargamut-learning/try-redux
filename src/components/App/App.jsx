import * as React from 'react';
import PostsList from '../PostsList/PostsListContainer';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.createPost = props.createPost;

    this.postTitleInput = React.createRef();
  }

  handleCreatePost = (event) => {
    event.preventDefault();

    this.createPost({ title: this.postTitleInput.current.value });
  };

  render() {
    const { text = `Default title`, image } = this.props;

    return (
      <>
        <h1>{this.text}</h1>

        <form onSubmit={this.handleCreatePost}>
          <input ref={this.postTitleInput} />

          <input type="submit" value="Create post" />
        </form>

        <PostsList />

        <img src={image} />
      </>
    );
  }
}

export default App;
