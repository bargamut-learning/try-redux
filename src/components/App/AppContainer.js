import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import createPost from '../../store/actionCreators/createPost';

import App from './App';

const mapStateToProps = (state, ownProps) => ({
  posts: state.posts,
});

const mapDispatchToProps = {
  createPost,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
