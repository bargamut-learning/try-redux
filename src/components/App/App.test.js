'use strict';

import * as React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import App from './App';

const mock = {
  text: `Test data`,
};

it('should render App correctly without data', () => {
  const renderer = new ShallowRenderer();

  renderer.render(<App />);

  const app = renderer.getRenderOutput();

  expect(app).toMatchSnapshot();
});

it('should render App correctly with mock title', () => {
  const { text: title } = mock;

  const renderer = new ShallowRenderer();

  renderer.render(<App text={title} />);

  const app = renderer.getRenderOutput();

  expect(app).toMatchSnapshot();
});
