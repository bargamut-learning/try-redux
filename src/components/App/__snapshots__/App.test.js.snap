// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App correctly with mock title 1`] = `
<React.Fragment>
  <h1>
    Test data
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input />
    <input
      type="submit"
      value="Create post"
    />
  </form>
  <Memo(Connect(PostsList)) />
  <img />
</React.Fragment>
`;

exports[`should render App correctly without data 1`] = `
<React.Fragment>
  <h1>
    Default title
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input />
    <input
      type="submit"
      value="Create post"
    />
  </form>
  <Memo(Connect(PostsList)) />
  <img />
</React.Fragment>
`;
